fuzz/fuzz_targets/header_length.rs

```diff
@@ -0,0 +1,28 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+use mongodb::cmap::conn::wire::{
+    header::{Header, OpCode},
+    message::Message,
+};
```

Comment:
I'd prefer not to add any of this to the public API, even if it's behind a feature flag. Can we move the contents of `fuzz` to within `src/test` and put any fuzzing additions to the message/header code behind `cfg(test)`?

Reply:
Unfortunately, this isn't possible because of the way cargo fuzz works (I wish it were, it's an issue with how it uses LLVM's fuzzing support, I think). However, you did bring to my attention that I was accidentally leaking some of this publicly when not fuzzing, so I have fixed that up.